### PR TITLE
[zh] Update Customizing DNS Service

### PR DESCRIPTION
**Pull request by ydFu**
* Update dns-custom-nameservers.md
   `content\zh-cn\docs\tasks\administer-cluster\dns-custom-nameservers.md`

* Synchronized English version and add an anchor. 

Signed-off-by: ydFu <ader.ydfu@gmail.com>

### CODE REVIEW

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 1]
```diff
@@ -87,51 +88,57 @@ DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记
...
 
 <!-- 
-If a Pod's `dnsPolicy` is set to "`default`", it inherits the name resolution
+If a Pod's `dnsPolicy` is set to `default`, it inherits the name resolution
 configuration from the node that the Pod runs on. The Pod's DNS resolution
 should behave the same as the node.
 But see [Known issues](/docs/tasks/administer-cluster/dns-debugging-resolution/#known-issues). 
 -->
-如果 Pod 的 `dnsPolicy` 设置为 "`default`"，则它将从 Pod 运行所在节点继承名称解析配置。
+如果 Pod 的 `dnsPolicy` 设置为 `default，则它将从 Pod 运行所在节点继承名称解析配置。
```
Comment (ydFu) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:96]:
```suggestion
如果 Pod 的 `dnsPolicy` 设置为 `default`，则它将从 Pod 运行所在节点继承名称解析配置。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 2]
```diff
@@ -87,51 +88,57 @@ DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记
...
 ### CoreDNS ConfigMap 选项  {#coredns-configmap-options}
 
 CoreDNS 是模块化且可插拔的 DNS 服务器，每个插件都为 CoreDNS 添加了新功能。
 可以通过维护 [Corefile](https://coredns.io/2017/07/23/corefile-explained/)，即 CoreDNS 配置文件，
-来定制其行为。 集群管理员可以修改 CoreDNS Corefile 的 ConfigMap，以更改服务发现的工作方式。
+来定制其行为。 
+
+作为集群管理员，可以修改
+{{<glossary_tooltip text="ConfigMap" term_id="configmap" >}}
+用于 CoreDNS Corefile 更改 DNS 服务发现方式该集群的行为。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:135]:
```suggestion
用于 CoreDNS Corefile 的 {{<glossary_tooltip text="ConfigMap" term_id="configmap" >}}
以更改 DNS 服务发现方式该集群的行为。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 3]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
-
-  <!-- 
-  The `pods insecure` option is provided for backward compatibility with kube-dns. You can use the
-  `pods verified` option, which returns an A record only if there exists a pod in same namespace
-  with matching IP. The `pods disabled` option can be used if you don't use pod records. 
-  -->
-  `pods insecure` 选项是为了与 kube-dns 向后兼容。你可以使用 `pods verified` 选项，该选项使得
-  仅在相同名称空间中存在具有匹配 IP 的 Pod 时才返回 A 记录。如果你不使用 Pod 记录，则可以使用
-  `pods disabled` 选项。
+* [health](https://coredns.io/plugins/health/)：CoreDNS 的健康报告给`http://localhost:8080/health`。在这个扩展语法中，`lameduck` 会使进程变得不健康，然后等待 5 秒，然后再关闭进程。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:182]:
```suggestion
* [health](https://coredns.io/plugins/health/)：在 `http://localhost:8080/health` 提供 CoreDNS 的健康报告。
  在这个扩展语法中，`lameduck` 会使进程变得不健康，然后等待 5 秒，然后再关闭进程。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 4]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+* [ready](https://coredns.io/plugins/ready/)：当所有能够发出就绪信号的插件都这样做时，端口 8181 上的 HTTP 端点将返回 200 OK。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:183]:
```suggestion
* [ready](https://coredns.io/plugins/ready/)：当所有能够发出就绪信号的插件都发出就绪信号时，
  端口 8181 上的 HTTP 端点将返回 200 OK。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 5]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+
+
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:185]: 删除多余空行

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 6]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+<!-- 
+* [kubernetes](https://coredns.io/plugins/kubernetes/): CoreDNS will reply to DNS queries based on IP of the services and pods of Kubernetes. You can find [more details](https://coredns.io/plugins/kubernetes/) about that plugin on the CoreDNS website. `ttl` allows you to set a custom TTL for responses. The default is 5 seconds. The minimum TTL allowed is 0 seconds, and the maximum is capped at 3600 seconds. Setting TTL to 0 will prevent records from being cached.  
+  The `pods insecure` option is provided for backward compatibility with _kube-dns_. You can use the `pods verified` option, which returns an A record only if there exists a pod in same namespace with matching IP. The `pods disabled` option can be used if you don't use pod records.
+-->
+* [kubernetes](https://coredns.io/plugins/kubernetes/)：CoreDNS 将基于 Kubernetes 的服务和 Pod 的 IP 回复 DNS 查询。你可以在 CoreDNS 网站上找到有关该插件的[更多详细信息](https://coredns.io/plugins/kubernetes/)。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:190]:
```suggestion
* [kubernetes](https://coredns.io/plugins/kubernetes/)：CoreDNS 将基于 Kubernetes
  的服务和 Pod 的 IP 回复 DNS 查询。你可以在 CoreDNS
  网站上找到有关该插件的[更多详细信息](https://coredns.io/plugins/kubernetes/)。 
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 7]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+* [ready](https://coredns.io/plugins/ready/)：当所有能够发出就绪信号的插件都这样做时，端口 8181 上的 HTTP 端点将返回 200 OK。
+
+
+<!-- 
+* [kubernetes](https://coredns.io/plugins/kubernetes/): CoreDNS will reply to DNS queries based on IP of the services and pods of Kubernetes. You can find [more details](https://coredns.io/plugins/kubernetes/) about that plugin on the CoreDNS website. `ttl` allows you to set a custom TTL for responses. The default is 5 seconds. The minimum TTL allowed is 0 seconds, and the maximum is capped at 3600 seconds. Setting TTL to 0 will prevent records from being cached.  
+  The `pods insecure` option is provided for backward compatibility with _kube-dns_. You can use the `pods verified` option, which returns an A record only if there exists a pod in same namespace with matching IP. The `pods disabled` option can be used if you don't use pod records.
+-->
+* [kubernetes](https://coredns.io/plugins/kubernetes/)：CoreDNS 将基于 Kubernetes 的服务和 Pod 的 IP 回复 DNS 查询。你可以在 CoreDNS 网站上找到有关该插件的[更多详细信息](https://coredns.io/plugins/kubernetes/)。 
+  `ttl` 允许你为响应设置自定义 TTL。默认值为 5 秒。允许的最小 TTL 为 0 秒，最大值为 3600 秒。将 TTL 设置为 0 将防止记录被缓存。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:191]:
```suggestion
  `ttl` 允许你为响应设置自定义 TTL。默认值为 5 秒。允许的最小 TTL 为 0 秒，最大值为 3600 秒。
  将 TTL 设置为 0 将防止记录被缓存。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 8]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+  提供 `pods insecure` 选项是为了向后兼容 _kube-dns_。你可以使用 `pods encrypted` 选项，该选项仅在同一命名空间中存在具有匹配 IP 的 pod 时才返回 A 记录。如果你不使用 pod 记录，则可以使用 `pods disabled` 选项。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:192]:
```suggestion
  提供 `pods insecure` 选项是为了向后兼容 _kube-dns_。你可以使用 `pods encrypted` 选项，
  该选项仅在同一命名空间中存在具有匹配 IP 的 Pod 时才返回 A 记录。
  如果你不使用 p、Pod 记录，则可以使用 `pods disabled` 选项。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 9]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
 * [cache](https://coredns.io/plugins/cache/): This enables a frontend cache.
 * [loop](https://coredns.io/plugins/loop/): Detects simple forwarding loops and halts the CoreDNS process if a loop is found.
 * [reload](https://coredns.io/plugins/reload): Allows automatic reload of a changed Corefile. After you edit the ConfigMap configuration, allow two minutes for your changes to take effect.
 * [loadbalance](https://coredns.io/plugins/loadbalance): This is a round-robin DNS loadbalancer that randomizes the order of A, AAAA, and MX records in the answer. 
 -->
-* [prometheus](https://coredns.io/plugins/prometheus/)：CoreDNS 的度量指标值以
-  [Prometheus](https://prometheus.io/) 格式在 http://localhost:9153/metrics 上提供。
-* [forward](https://coredns.io/plugins/forward/): 不在 Kubernetes 集群域内的任何查询都将转发到
-  预定义的解析器 (/etc/resolv.conf).
+* [prometheus](https://coredns.io/plugins/metrics/)：CoreDNS 的指标可在 [Prometheus](https://prometheus.io/) 中的 `http://localhost:9153/metrics` 获得格式（也称为 OpenMetrics）。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:202]:
```suggestion
* [prometheus](https://coredns.io/plugins/metrics/)：CoreDNS 的指标可在
  `http://localhost:9153/metrics` 获得，其数据格式（也称为 OpenMetrics）
  可在 [Prometheus](https://prometheus.io/) 网站获得。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 10]
```diff
@@ -168,39 +175,33 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+* [forward](https://coredns.io/plugins/forward/)：不在 Kubernetes 集群域内的任何查询都将转发到
+  预定义的解析器 (/etc/resolv.conf)。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:204]:
```suggestion
* [forward](https://coredns.io/plugins/forward/)：不在 Kubernetes
  集群域内的任何查询都将转发到预定义的解析器 (/etc/resolv.conf)。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 11]
```diff
@@ -218,7 +219,7 @@ You can modify the default CoreDNS behavior by modifying the ConfigMap.
 
 CoreDNS has the ability to configure stubdomains and upstream nameservers using the [forward plugin](https://coredns.io/plugins/forward/).
 -->
-### 使用 CoreDNS 配置存根域和上游域名服务器
+### 使用 CoreDNS 配置存根域和上游域名服务器 {#configuration-of-stub-domain-and-upstream-nameserver-using-coredns}
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:222]: 加这种 anchor 意义不大，没有页面会引用这么长的 anchor

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 12]
```diff
@@ -286,15 +287,17 @@ data:
 ```
 
 <!-- 
-The `kubeadm` supports automatic translation of the CoreDNS ConfigMap from the kube-dns ConfigMap.
+The `kubeadm` tool supports automatic translation from the kube-dns ConfigMap
+to the equivalent CoreDNS ConfigMap.
 -->
-工具 `kubeadm` 支持将 kube-dns ConfigMap 自动转换为 CoreDNS ConfigMap。
+`kubeadm` 工具支持从 kube-dns ConfigMap 
+自动转换为到等效的 CoreDNS ConfigMap。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:294]:
```suggestion
`kubeadm` 工具支持将 kube-dns ConfigMap 自动转换为等效的 CoreDNS ConfigMap。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 13]
```diff
@@ -303,13 +306,13 @@ During translation, all FQDN nameservers will be omitted from the CoreDNS config
...
 CoreDNS supports the features of kube-dns and more.
-A ConfigMap created for kube-dns to support `StubDomains`and `upstreamNameservers` translates to the `proxy` plugin in CoreDNS.
+A ConfigMap created for kube-dns to support `StubDomains`and `upstreamNameservers` translates to the `forward` plugin in CoreDNS.
 
 ### Example
 
 This example ConfigMap for kube-dns specifies stubdomains and upstreamnameservers: 
 -->
-## CoreDNS 配置等同于 kube-dns
+## CoreDNS 配置等同于 kube-dns{#coredns-configuration-equivalent-to-kube-dns}
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:315]:
```suggestion
## CoreDNS 配置等同于 kube-dns    {#coredns-configuration-equivalent-to-kube-dns}
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 14]
```diff
@@ -330,23 +333,24 @@ kind: ConfigMap
...
+    errors
+    cache 30
+    forward . 1.2.3.4
+}
+my.cluster.local:53 {
+    errors
+    cache 30
+    forward . 2.3.4.5
+}
+```
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:353]: 现在的缩进是对的。

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 15]
```diff
@@ -383,21 +387,27 @@ my.cluster.local:53 {
...
+is available to help users adapt CoreDNS in place of kube-dns.
+
+You can also migrate using the official CoreDNS
+[deploy script](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh).
 
 -->
-## 迁移到 CoreDNS
+## 迁移到 CoreDNS {#migration-to-coreDNS}
+
+从 kube-dns 迁移到 CoreDNS，详细
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:400]:
```suggestion
要从 kube-dns 迁移到 CoreDNS，可参阅详细[博客文章](https://coredns.io/2018/05/21/migration-from-kube-dns-to-coredns/)；
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 16]
```diff
@@ -383,21 +387,27 @@ my.cluster.local:53 {
...
+[博客文章](https://coredns.io/2018/05/21/migration-from-kube-dns-to-coredns/)
+可用于帮助用户适应 CoreDNS 代替 kube-dns。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:402]:
```suggestion
该文章可帮助用户定制 CoreDNS 以代替 kube-dns。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 17]
```diff
@@ -383,21 +387,27 @@ my.cluster.local:53 {
...
 
-要从 kube-dns 迁移到 CoreDNS，[此博客](https://coredns.io/2018/05/21/migration-from-kube-dns-to-coredns/)
-提供了帮助用户将 kube-dns 替换为 CoreDNS。
-集群管理员还可以使用[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)
-进行迁移。
+你也可以使用官方 CoreDNS 进行迁移
+[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:405]:
```suggestion
你也可以使用官方[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)进行CoreDNS 迁移。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 14 above)
Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:353]: Here is now synchronized with English

Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:360]: Could the tech lead recommend starting the changes upstream, so I can start a PR to make the corrections.

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 18]
```diff
@@ -87,51 +88,57 @@ DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记
...
+behaves for that cluster. 
 -->
 ### CoreDNS ConfigMap 选项  {#coredns-configmap-options}
 
 CoreDNS 是模块化且可插拔的 DNS 服务器，每个插件都为 CoreDNS 添加了新功能。
 可以通过维护 [Corefile](https://coredns.io/2017/07/23/corefile-explained/)，即 CoreDNS 配置文件，
-来定制其行为。 集群管理员可以修改 CoreDNS Corefile 的 ConfigMap，以更改服务发现的工作方式。
+来定制其行为。 
+
+作为集群管理员，可以修改
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:133]:
```suggestion
作为集群管理员，可以修改用于
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 19]
```diff
@@ -87,51 +88,57 @@ DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记
...
+用于 CoreDNS Corefile 的 {{<glossary_tooltip text="ConfigMap" term_id="configmap" >}}
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:134]:
```suggestion
CoreDNS Corefile 的 {{<glossary_tooltip text="ConfigMap" term_id="configmap" >}}
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 20]
```diff
@@ -87,51 +88,57 @@ DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记
...
+以更改 DNS 服务发现方式该集群的行为。
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:135]:
```suggestion
以更改针对该集群时 DNS 服务发现的行为方式。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 21]
```diff
@@ -168,39 +175,41 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
-  <!-- 
-  The `pods insecure` option is provided for backward compatibility with kube-dns. You can use the
-  `pods verified` option, which returns an A record only if there exists a pod in same namespace
-  with matching IP. The `pods disabled` option can be used if you don't use pod records. 
-  -->
-  `pods insecure` 选项是为了与 kube-dns 向后兼容。你可以使用 `pods verified` 选项，该选项使得
-  仅在相同名称空间中存在具有匹配 IP 的 Pod 时才返回 A 记录。如果你不使用 Pod 记录，则可以使用
-  `pods disabled` 选项。
+* [health](https://coredns.io/plugins/health/)：在 `http://localhost:8080/health` 提供 CoreDNS 的健康报告。
+  在这个扩展语法中，`lameduck` 会使进程变得不健康，然后等待 5 秒，然后再关闭进程。
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:183]:
```suggestion
  按照这个扩展的语法，`lameduck` 会使进程变得不健康，等待 5 秒后再关闭该进程。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 22]
```diff
@@ -168,39 +175,41 @@ Corefile 配置包括以下 CoreDNS [插件](https://coredns.io/plugins/)：
...
+* [kubernetes](https://coredns.io/plugins/kubernetes/): CoreDNS will reply to DNS queries based on IP of the services and pods of Kubernetes. You can find [more details](https://coredns.io/plugins/kubernetes/) about that plugin on the CoreDNS website. `ttl` allows you to set a custom TTL for responses. The default is 5 seconds. The minimum TTL allowed is 0 seconds, and the maximum is capped at 3600 seconds. Setting TTL to 0 will prevent records from being cached.  
+  The `pods insecure` option is provided for backward compatibility with _kube-dns_. You can use the `pods verified` option, which returns an A record only if there exists a pod in same namespace with matching IP. The `pods disabled` option can be used if you don't use pod records.
+-->
+* [kubernetes](https://coredns.io/plugins/kubernetes/)：CoreDNS 将基于 Kubernetes
+  的服务和 Pod 的 IP 回复 DNS 查询。你可以在 CoreDNS
+  网站上找到有关该插件的[更多详细信息](https://coredns.io/plugins/kubernetes/)。 
+  `ttl` 允许你为响应设置自定义 TTL。默认值为 5 秒。允许的最小 TTL 为 0 秒，最大值为 3600 秒。
+  将 TTL 设置为 0 将防止记录被缓存。
+  提供 `pods insecure` 选项是为了向后兼容 _kube-dns_。你可以使用 `pods encrypted` 选项，
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:196]:
```suggestion
  提供 `pods insecure` 选项是为了向后兼容 **kube-dns**。你可以使用 `pods encrypted` 选项，
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 20 above)
Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:135]: This section has been adjusted as suggested by @tengqm

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 22 above)
Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:196]: English does not need to be adjusted

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 17 above)
Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md]: Thank for code view！

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 18 above)
Comment (ydFu) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md]: Thank for code view!

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 23]
```diff
@@ -84,54 +85,60 @@ The DNS server supports forward lookups (A and AAAA records), port lookups (SRV
 and more. For more information, see [DNS for Services and Pods](/docs/concepts/services-networking/dns-pod-service/).
 -->
 DNS 服务器支持正向查找（A 和 AAAA 记录）、端口发现（SRV 记录）、反向 IP 地址发现（PTR 记录）等。
-更多信息，请参见[Pod 和 服务的 DNS](/zh-cn/docs/concepts/services-networking/dns-pod-service/)。
+更多信息，请参见 [Pod 和 服务的 DNS](/zh-cn/docs/concepts/services-networking/dns-pod-service/)。
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:88]:
```suggestion
更多信息，请参见 [Pod 与 Service 的 DNS](/zh-cn/docs/concepts/services-networking/dns-pod-service/)。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 24]
```diff
@@ -383,21 +394,25 @@ my.cluster.local:53 {
...
+## 迁移到 CoreDNS {#migration-to-coreDNS}
+
+要从 kube-dns 迁移到 CoreDNS，参阅详细[博客文章](https://coredns.io/2018/05/21/migration-from-kube-dns-to-coredns/)；
+该文章可帮助用户定制 CoreDNS 以代替 kube-dns。
 
-要从 kube-dns 迁移到 CoreDNS，[此博客](https://coredns.io/2018/05/21/migration-from-kube-dns-to-coredns/)
-提供了帮助用户将 kube-dns 替换为 CoreDNS。
-集群管理员还可以使用[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)
-进行迁移。
+你也可以使用官方[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)进行CoreDNS 迁移。
```
Comment (windsonsea) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:410]:
```suggestion
你也可以使用官方[部署脚本](https://github.com/coredns/deployment/blob/master/kubernetes/deploy.sh)进行 CoreDNS 迁移。
```

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 14 above)
Comment (tengqm) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:360]: https://github.com/kubernetes/website/pull/35923 is proposed.

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md [hunk 25]
```diff
@@ -33,19 +33,20 @@ explains how to use `kubeadm` to migrate from `kube-dns`.
...
-[cluster add-on](http://releases.k8s.io/master/cluster/addons/README.md). 
+using the _addon manager_
+[cluster add-on](http://releases.k8s.io/master/cluster/addons/README.md).
 -->
-## 介绍
+## 介绍 {#introduction}
 
-DNS 是使用[集群插件](http://releases.k8s.io/master/cluster/addons/README.md)
-管理器自动启动的内置的 Kubernetes 服务。
+DNS 是使用 **插件管理器** [cluster add-on](http://releases.k8s.io/master/cluster/addons/README.md)自动启动的内置的 Kubernetes 服务。
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:49]: cluster add-on 为什么改回去？

File: content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md (hunk 20 above)
Comment (tengqm) [reply content/zh-cn/docs/tasks/administer-cluster/dns-custom-nameservers.md:135]: @windsonsea 的语序更好些。我的建议只针对前半句。

**Comment by Sea-n** [comment]
Hey @ydFu,

It's been a while since the last update, would you like to address issues in the previous comments so that we can move forward?

**Comment by Sea-n** [comment]
Hey @ydFu,

Do you have time to address the above issues? It's been 3 weeks since last update.

**Comment by Sea-n** [comment]
Note: Replaced by PR #36991.
- #36991

**Pull request closed without merging**